### PR TITLE
Refactor: WindowsAuthImpersonation can be used in unit tests (when Ht…

### PR DESCRIPTION
**Pull request by damirdean**
…tpContext is not available) with a HttpContextAccessorMock

### CODE REVIEW

File: Build.bat [hunk 1]
```diff
@@ -2,8 +2,17 @@ SETLOCAL
 SET Version=1.1.0
 SET Prerelease=auto
 
-IF NOT DEFINED VisualStudioVersion CALL "%VS140COMNTOOLS%VsDevCmd.bat" || ECHO ERROR: Cannot find Visual Studio 2015, missing VS140COMNTOOLS variable. && GOTO Error0
+@IF DEFINED VisualStudioVersion GOTO SkipVcvarsall
+@SET VSTOOLS=
+@IF EXIST "%VS100COMNTOOLS%..\..\VC\vcvarsall.bat" SET VSTOOLS="%VS100COMNTOOLS%..\..\VC\vcvarsall.bat" x86
```
Comment (bantolov) [inline Build.bat]: Please remove older version from VS100 to VS130. Build will not work on those versions of Visual Studio.

File: Source/Rhetos.WindowsAuthImpersonation/HttpContextAccessorMock.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Web.Security;
+using Rhetos.WindowsAuthImpersonation.Abstractions;
+
+namespace Rhetos.WindowsAuthImpersonation
+{
+    public class HttpContextAccessorMock : IHttpContextAccessor
```
Comment (bantolov) [inline Source/Rhetos.WindowsAuthImpersonation/HttpContextAccessorMock.cs]: This class is not used in the project, should be removed.

File: Source/Rhetos.WindowsAuthImpersonation/ImpersonationService.cs [hunk 3]
```diff
@@ -232,20 +223,21 @@ private void SetImpersonatedUser(string impersonatedUser)
         {
             if (string.IsNullOrEmpty(impersonatedUser))
             {
-                TicketUtility.AddToResponseCookie(null, _httpContextAccessor.HttpContext);
```
Comment (bantolov) [inline Source/Rhetos.WindowsAuthImpersonation/ImpersonationService.cs:235]: Delete TicketUtility.AddToResponseCookie (if it is not used anymore)

File: Source/Rhetos.WindowsAuthImpersonation/ImpersonationUserInfo.cs [hunk 4]
```diff
@@ -20,57 +20,45 @@
...
     [Export(typeof(IUserInfo))]
     public class ImpersonationUserInfo : IImpersonationUserInfo
     {
+        private readonly ImpersonationService _impersonationService;
+
         #region IUserInfo implementation
 
-        public bool IsUserRecognized => _isUserRecognized.Value;
-        public string UserName => _impersonatedUser.Value ?? _actualUser.Value;
+        public bool IsUserRecognized => _impersonationService.GetActualUserName() != null;
```
Comment (bantolov) [inline Source/Rhetos.WindowsAuthImpersonation/ImpersonationUserInfo.cs]: Use string.IsNullOrEmpty

**Review by bantolov — CHANGES_REQUESTED** [review]

File: Source/Rhetos.WindowsAuthImpersonation/ImpersonationService.cs (hunk 3 above)
Comment (damirdean) [reply Source/Rhetos.WindowsAuthImpersonation/ImpersonationService.cs:235]: It is used twice in ImpersonationTicketExpirationModule class

**Review by bantolov — APPROVED** [review]

**Pull request merged**